pkg/actuators/machine/instances.go

```diff
@@ -201,22 +203,34 @@ func getBlockDeviceMappings(blockDeviceMappings []providerconfigv1.BlockDeviceMa
 	}
 	deviceName := describeAMIResult.Images[0].RootDeviceName
 
-	// Only support one blockDeviceMapping
-	volumeSize := blockDeviceMappings[0].EBS.VolumeSize
-	volumeType := blockDeviceMappings[0].EBS.VolumeType
-	blockDeviceMapping := ec2.BlockDeviceMapping{
-		DeviceName: deviceName,
-		Ebs: &ec2.EbsBlockDevice{
-			VolumeSize: volumeSize,
-			VolumeType: volumeType,
-			Encrypted:  blockDeviceMappings[0].EBS.Encrypted,
-		},
-	}
-	if *volumeType == "io1" {
-		blockDeviceMapping.Ebs.Iops = blockDeviceMappings[0].EBS.Iops
+	for _, blockDeviceMappingSpec := range blockDeviceMappingSpecs {
+		if blockDeviceMappingSpec.EBS == nil {
+			continue
+		}
+
+		if *blockDeviceMappingSpec.DeviceName != "" {
+			deviceName = blockDeviceMappingSpec.DeviceName
+		}
+
+		volumeSize := blockDeviceMappingSpec.EBS.VolumeSize
+		volumeType := blockDeviceMappingSpec.EBS.VolumeType
+
+		blockDeviceMapping := ec2.BlockDeviceMapping{
+			DeviceName: deviceName,
+			Ebs: &ec2.EbsBlockDevice{
+				VolumeSize: volumeSize,
+				VolumeType: volumeType,
+				Encrypted:  blockDeviceMappingSpec.EBS.Encrypted,
+			},
+		}
+		if *volumeType == "io1" {
```

Comment:
should we have this `"io1"` value a constant somewhere so it doesn't seem like a _magic value_ ?

Reply:
good question :)